### PR TITLE
Move balance parse logic into parseBalance

### PR DESCRIPTION
**Pull request by fernandocalsa**
Related with https://github.com/ccxt/ccxt/issues/10960

I move the parsing logic for each exchange to a specific function, so it is easier to follow.

This is only an example with Binance, if you think this is ok I will apply the same change to every exchange so everything is aligned

### CODE REVIEW

**Comment by frosty00** [comment]
why is it called normalizeBalance and why does it not return anything

**Comment by fernandocalsa** [comment]
@frosty00 sorry, I did it quickly, I even forgot passing the `type`.
Fixed in my last commit https://github.com/ccxt/ccxt/pull/10980/commits/a5af28fd03f365f35c9d0fd845c697e5c53d67dc

File: js/binance.js [hunk 1]
```diff
@@ -1401,6 +1401,69 @@ module.exports = class binance extends Exchange {
...
+                const balance = balances[i];
+                const currencyId = this.safeString (balance, 'asset');
+                const code = this.safeCurrencyCode (currencyId);
+                const account = this.account ();
+                account['free'] = this.safeString (balance, 'availableBalance');
+                account['used'] = this.safeString (balance, 'initialMargin');
+                account['total'] = this.safeString2 (balance, 'marginBalance', 'balance');
+                result[code] = account;
+            }
+        }
```
Comment (mrbaguvix) [inline js/binance.js:1461]: A switch/case would be cleaner right?

File: js/binance.js [hunk 2]
```diff
@@ -1401,6 +1401,69 @@ module.exports = class binance extends Exchange {
...
+                const withdrawing = this.safeString (entry, 'withdrawing');
+                const locked = this.safeString (entry, 'locked');
+                account['used'] = Precise.stringAdd (frozen, Precise.stringAdd (locked, withdrawing));
+                result[code] = account;
+            }
+        } else {
+            let balances = response;
+            if (!Array.isArray (response)) {
+                balances = this.safeValue (response, 'assets', []);
+            }
```
Comment (mrbaguvix) [inline js/binance.js:1450]: Something doesn't seem right here

**Review by mrbaguvix — COMMENTED** [review]
Thanks for the revamp!

File: js/binance.js (hunk 1 above)
Comment (kroitor) [reply js/binance.js:1461]: Probably, yes, but we don't transpile switch-case clauses: 

- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#multilanguage-support
- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#derived-exchange-classes

File: js/binance.js (hunk 2 above)
Comment (kroitor) [reply js/binance.js:1450]: @mrbaguvix what do you mean exactly? Can you please elaborate?

**Pull request merged**